lib/galaxy/jobs/runners/util/job_script/__init__.py

```diff
@@ -96,6 +96,10 @@ def check_script_integrity(config):
 
 
 def write_script(path, contents, config, mode=0o755):
+    dir = os.path.dirname(path)
+    if not os.path.exists(dir):
```

Comment:
No worry of contention here - so it isn't really needed IMO. I don't know if we should start globally replacing os.makedirs with safe_makedirs... I'm going to vote for no I guess.